doc/odoc_for_authors.mld

```diff
@@ -663,6 +663,29 @@ v}
 In this case the reference text would be "Foo Bar" so the paragraph would read
 "See Foo Bar for details".
 
+Note that label names cannot contain [-], you should use [_] instead.
```

Comment:
It's possible !
However, the reference syntax is particular:
```
  {!"that-section"}
  {!that_other_page."that-section"}
  {!section-"that-section"}
```

Reply:
Do you want me to document that?

That looks a bit like some sort of abuse :stuck_out_tongue:.

If you want this documented I think it'd be nice to have some terminology to distinguish the two types of labels. Something like "plain" vs "quoted". I think I'd also need to know what characters can be used in either forms. I'm assuming that inside quoted labels you can put whatever you want except unescaped `"`, although I'm not sure why one would want to use such characters for a label, hence me questioning documenting this in the first place.

Let me know what you prefer!

Reply:
It's not the place to document that in details but I think a little hint for reminding the reference syntax could be useful. 

The quotes are part of the reference syntax so it's more general than just labels but it's also quircky: it exists because the `-` is used for specifying the "kind" of earch segment of the reference, and there are "kinds" that contains it too (`module-type-Foo`). Also, there's no syntax for escaping quotes, so it's really just about the `-`.

Reply:
Ok I'll update the corresponding sections and this one as well!